bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -99,21 +97,22 @@ public IntesisHomeHandler(final Thing thing, final HttpClient httpClient,
     @Override
     public void initialize() {
         updateStatus(ThingStatus.UNKNOWN);
-        final IntesisConfiguration config = getConfigAs(IntesisConfiguration.class);
-        ipAddress = config.ipAddress;
-        password = config.password;
-        if (ipAddress.isEmpty() || password.isEmpty()) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        config = getConfigAs(IntesisConfiguration.class);
+        if (config.ipAddress.isEmpty() && config.password.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "IP-Address and password not set");
+        } else if (config.ipAddress.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "IP-Address not set");
+        } else if (config.password.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Password not set");
         }
```

Comment:
Once you find that the configuration is invalid, you should bail out early subsequent errors from the improper configuration would end up burying the initial config errors.
```suggestion
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "IP-Address and password not set");
            return;
        } else if (config.ipAddress.isEmpty()) {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "IP-Address not set");
            return;
        } else if (config.password.isEmpty()) {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Password not set");
            return;
        }
```

Reply:
Maybe better to add an else case at the last if and put the rest of the code in there
```
 } else {
    // start background initialization:
   ....
}
```

Reply:
Done, implemented both changes.

Reply:
With the else the return is technically not needed :wink: :